src/include/migraphx/op/pooling.hpp

```diff
@@ -390,7 +411,15 @@ struct pooling
                     // this is a padding element.  Padding locations
                     // don't contribute to average or max pooling total but can play in
                     // lpnorm pooling.
-                    output_val = op(output_val, 0);
```

Comment:
This should just be `output_val = op(output_val, op.template init<Type>());`. Using `0` is a bug.